poly-synth/src/noisegate-side-chain.js

```diff
@@ -0,0 +1,212 @@
+ * limitations under the License.
+ */
+
+/**
+ * @class NoiseGateSideChain
+ * @description This script-processor-wrapper class is replica of
+ * audio-worklet/src/noise-gates-script-processor.js. NoiseGateSideChain builds
+ * an envelope on the the input from the first channel and returns the output of
+ * the second input channel with the samples muted (according to attack and
+ * release) when the level in the first channel registers above the specified
+ * threshold. This modification always processes stereo input and returns mono
+ * output.
+ */
+class NoiseGateSideChain {
+  /**
+   * @constructor
+   * @param {BaseAudioContext} context The audio context
+   * @param {Object} options Parameters for the noise gate.
+   * @param {Number} options.attack Seconds for gate to fully close. The default
+   *                                is zero.
+   * @param {Number} options.release Seconds for gate to fully open. The default
+   *                                 is zero.
+   * @param {Number} options.bufferSize The size of an onaudioprocess window.
+   *                                    The default lets the script processor
+   *                                    decide.
+   * @param {Number} options.timeConstant Seconds for envelope follower's
+   *                                      smoothing filter delay. The
+   *                                      default has been set experimentally to
+   *                                      0.0025s.
+   * @param {Number} options.threshold Decibel level beneath which sound is
+   *                                   muted in the second channel.
+   *                                   The default is -100 dBFS.
+   */
+  constructor(context, options) {
+    if (!(context instanceof BaseAudioContext))
+      throw 'Not a valid audio context.';
+    if (!options) options = {};
+
+    this.context_ = context;
+    const bufferSize = options.bufferSize || 0;
+    this.attack = options.attack || 0;
+    this.release = options.release || 0;
+
+    // The time constant of the filter has been set experimentally to balance
+    // roughly delay for high frequency suppression.
+    let timeConstant = (typeof options.timeConstant === 'undefined') ?
+        0.0025 : options.timeConstant;
+
+    this.threshold = (typeof options.threshold === 'undefined') ?
+        -100 : options.threshold;
+
+    // Alpha controls a tradeoff between the smoothness of the
+    // envelope and its delay, with a higher value giving more smoothness at
+    // the expense of delay and vice versa.
+    this.alpha_ = this.getAlphaFromTimeConstant_(
+        timeConstant, this.context_.sampleRate);
+
+    this.noiseGateKernel_ = this.context_.createScriptProcessor(
+        bufferSize, 2, 1);
+    this.noiseGateKernel_.onaudioprocess = this.onaudioprocess_.bind(this);
+
+    // The noise gate is connected to and from by dummy input and output nodes.
+    this.input = new GainNode(this.context_);
```

Comment:
@hoch and I discussed this, and decided the easiest solution would be to let the input have two channels and use one channel for the envelope and modify the contents of the second channel. 

I'm not sure how an implementation would work for the two node solution you describe.